nixos/modules/services/networking/wasabibackend.nix

```diff
@@ -0,0 +1,145 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.wasabibackend;
+
+  inherit (lib) mkEnableOption mkIf mkOption optionalAttrs types;
+
+	confOptions = {
+			BitcoinRpcConnectionString = "${cfg.rpc.user}:${cfg.rpc.password}";
+	} // optionalAttrs (cfg.network == "mainnet") {
+			Network = "Main";
+			MainNetBitcoinP2pEndPoint = "${cfg.endpoint.ip}:${toString cfg.endpoint.port}";
+			MainNetBitcoinCoreRpcEndPoint = "${cfg.rpc.ip}:${toString cfg.rpc.port}";
+	} // optionalAttrs (cfg.network == "testnet") {
+			Network = "TestNet";
+			TestNetBitcoinP2pEndPoint = "${cfg.endpoint.ip}:${toString cfg.endpoint.port}";
+			TestNetBitcoinCoreRpcEndPoint = "${cfg.rpc.ip}:${toString cfg.rpc.port}";
+	} // optionalAttrs (cfg.network == "regtest") {
+			Network = "RegTest";
+			RegTestBitcoinP2pEndPoint = "${cfg.endpoint.ip}:${toString cfg.endpoint.port}";
+			RegTestBitcoinCoreRpcEndPoint = "${cfg.rpc.ip}:${toString cfg.rpc.port}";
+	};
+
+	configFile = pkgs.writeText "wasabibackend.conf" (builtins.toJSON confOptions);
+
+in {
+
+  options = {
+
+    services.wasabibackend = {
+      enable = mkEnableOption "Wasabi backend service";
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/wasabibackend";
+        description = "The data directory for the Wasabi backend node.";
+      };
+
+			customConfigFile = mkOption {
+				type = types.nullOr types.path;
+				default = null;
+				description = "Defines the path to a custom configuration file that is copied to the user's directory. Overrites any config options.";
+			};
+
+      network = mkOption {
+        type = types.enum [ "mainnet" "testnet" "regtest" ];
+        default = "mainnet";
+        description = "The network to use for the Wasabi backend service.";
+      };
+
+			endpoint = {
+				ip = mkOption {
+					type = types.str;
+					default = "127.0.0.1";
+					description = "IP address for P2P connection to bitcoind.";
+				};				
+				port = mkOption {
+					type = types.port;
+					default = 18333;
+					description = "Port for P2P connection to bitcoind.";
+				};
+			};
+
+      rpc = {
+				ip = mkOption {
+					type = types.str;
+					default = "127.0.0.1";
+					description = "IP address for RPC connection to bitcoind.";
+				};
+
+				port = mkOption {
+					type = types.port;
+					default = 18332;
+					description = "Port for RPC connection to bitcoind.";
+				};
+
+        user = mkOption {
+          type = types.str;
+          default = "bitcoin";
+          description = "RPC user for the bitcoin endpoint.";
+        };
+
+        password = mkOption {
+          type = types.str;
+          default = "password";
+          description = "RPC password for the bitcoin endpoint. Warning: this is stored in cleartext in the Nix store! Use <literal>configFile</literal> if needed.";
+        };
```

Comment:
Since you already use `preStart` can you declare `passwordFile` instead and then use e.g. `jq` to add the password to the generated config?

Reply:
I solved it by proposing `customConfigFile` as there is many more configuration which are sadly undocumented. And given Wasabi backend is not using SSL for RPC, that password is not very safe in the first place.

Reply:
I did add `passwordFile` as well.